File: Java/BottomNavigationViewJava/app/src/main/java/com/example/bottomnavigationviewjava/MainActivity.java
package com.example.bottomnavigationviewjava;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;

import com.example.bottomnavigationviewjava.databinding.ActivityMainBinding;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private FragmentManager fragmentManager = getSupportFragmentManager();
    private RestaurantFragment restaurantFragment = new RestaurantFragment();
    private CheckFragment checkFragment = new CheckFragment();
    private NotifyFragment notifyFragment = new NotifyFragment();
    private MypageFragment mypageFragment = new MypageFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        FragmentTransaction transaction;
        transaction=fragmentManager.beginTransaction();
        transaction.replace(R.id.frameLayout, restaurantFragment).commitNowAllowingStateLoss();

        binding.bottomNavigationView.setOnItemSelectedListener(new BottomNavigationView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {

                FragmentTransaction transaction = fragmentManager.beginTransaction();

                int id = item.getItemId();

                switch (id){
                    case R.id.menu_restaurant:
                        transaction.replace(R.id.frameLayout, restaurantFragment).commitAllowingStateLoss();
                        break;
                    case R.id.menu_check:
                        transaction.replace(R.id.frameLayout, checkFragment).commitAllowingStateLoss();
                        break;
                    case R.id.menu_notify:
                        transaction.replace(R.id.frameLayout, notifyFragment).commitAllowingStateLoss();
                        break;
                    case R.id.menu_mypage:
                        transaction.replace(R.id.frameLayout, mypageFragment).commitAllowingStateLoss();
                        break;

                }
                return true;
            }
        });
    }
}

File: BottomNavigationViewKotlin/app/src/main/java/com/example/bottomnavigationviewkotlin/MainActivity.kt
package com.example.bottomnavigationviewkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.bottomnavigationviewkotlin.databinding.ActivityMainBinding
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var mealFragment: MealFragment
    private lateinit var applyFragment: ApplyFragment
    private lateinit var notifyFragment: NotifyFragment
    private lateinit var mypageFragment: MypageFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.bottomNav.setOnNavigationItemSelectedListener(onBottomNavigationSelectedListener)
    }

    private val onBottomNavigationSelectedListener = BottomNavigationView.OnNavigationItemReselectedListener{

        when(it.itemId){
            R.id.menu_meal ->{
                mealFragment = MealFragment.newInstatnce()
                supportFragmentManager.beginTransaction().replace(R.id.fragments_frame, mealFragment).commit()
            }
            R.id.menu_apply ->{
                applyFragment = ApplyFragment.newInstatnce()
                supportFragmentManager.beginTransaction().replace(R.id.fragments_frame, mealFragment).commit()
            }
            R.id.menu_notify ->{
                notifyFragment = NotifyFragment.newInstatnce()
                supportFragmentManager.beginTransaction().replace(R.id.fragments_frame, mealFragment).commit()
            }
            R.id.menu_mypage ->{
                mypageFragment = MypageFragment.newInstatnce()
                supportFragmentManager.beginTransaction().replace(R.id.fragments_frame, mealFragment).commit()
            }
        }
        true
    }
}

